### ISSUE TITLE
Unable to deploy from GitHub Package Registry

### ISSUE THREAD

**Issue by mvz** [getkuby/kuby-core#38]
I'm getting the following error when trying to deploy:

```sh
$ bundle exec kuby -e production deploy
error: bad URI(is not URI?): "GitHub Package Registry"
```

With debugging, I get:
```sh
$ GLI_DEBUG=true bundle exec kuby -e production deploy
error: bad URI(is not URI?): "GitHub Package Registry"
bundler: failed to load command: kuby (/home/matijs/.rbenv/versions/2.7.1/bin/kuby)
URI::InvalidURIError: bad URI(is not URI?): "GitHub Package Registry"
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/2.7.0/uri/rfc3986_parser.rb:67:in `split'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/2.7.0/uri/rfc3986_parser.rb:73:in `parse'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/2.7.0/uri/common.rb:234:in `parse'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/docker-remote-0.1.0/lib/docker/remote/client.rb:45:in `token'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/docker-remote-0.1.0/lib/docker/remote/client.rb:93:in `block in make_get'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/docker-remote-0.1.0/lib/docker/remote/client.rb:92:in `tap'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/docker-remote-0.1.0/lib/docker/remote/client.rb:92:in `make_get'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/docker-remote-0.1.0/lib/docker/remote/client.rb:21:in `tags'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/remote_tags.rb:28:in `tags'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/remote_tags.rb:39:in `timestamp_tags'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/tags.rb:68:in `timestamp_tags'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/tags.rb:73:in `latest_timestamp_tag'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/spec.rb:185:in `block in tag'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/spec.rb:184:in `fetch'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/docker/spec.rb:184:in `tag'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/kubernetes/spec.rb:75:in `before_deploy'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/kubernetes/spec.rb:106:in `deploy'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/tasks.rb:78:in `deploy'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/commands.rb:153:in `block (2 levels) in <class:Commands>'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/gli-2.19.2/lib/gli/command_support.rb:131:in `execute'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/gli-2.19.2/lib/gli/app_support.rb:296:in `block in call_command'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/gli-2.19.2/lib/gli/app_support.rb:309:in `call_command'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/gli-2.19.2/lib/gli/app_support.rb:83:in `run'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/lib/kuby/commands.rb:31:in `run'
  /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/kuby-core-0.11.8/bin/kuby:4:in `<top (required)>'
  /home/matijs/.rbenv/versions/2.7.1/bin/kuby:23:in `load'
  /home/matijs/.rbenv/versions/2.7.1/bin/kuby:23:in `<top (required)>'
```

**Comment by lazyatom** [comment]
I've had this error too. I believe it's because the `docker-remote` gem expects to find a URI in the "Realm" part of the authentication response, which is true for GitLab, but not GitHub.

**Comment by camertron** [comment]
Oh interesting, I was under the impression that's part of the Docker spec. I'll take a look at it.

**Comment by camertron** [comment]
@mvz I'm not able to reproduce this. The "realm" param in the authentication response contained a valid URL when I was testing just now (https://ghcr.io/token). Can you share your kuby config?

**Comment by mvz** [comment]
I'm using GitHub's package registry, not their new container registry. This is mainly because the container registry doesn't support using `GITHUB_TOKEN` for access from a GitHub Action yet. Here's my `kuby.rb`:

```ruby
# frozen_string_literal: true

require "dotenv/load"
require "kuby/linode"

# Define a production Kuby deploy environment
Kuby.define("PlayDate") do
  environment(:production) do
    docker do
      # Configure your Docker registry credentials here.
      credentials do
        username ENV["KUBY_DOCKER_USERNAME"]
        password ENV["KUBY_DOCKER_PASSWORD"]
        email ENV["KUBY_DOCKER_EMAIL"]
      end

      distro :alpine

      image_url "docker.pkg.github.com/mvz/playdate/kuby-test"
    end

    kubernetes do
      # Add a plugin that facilitates deploying a Rails app.
      add_plugin :rails_app do
        hostname "playdate-demo.matijs.net"
        # configure database credentials
        database do
          user ENV["KUBY_DB_USER"]
          password ENV["KUBY_DB_PASSWORD"]
        end
      end

      # Use Linode as the provider
      provider :linode do
        access_token ENV["KUBY_LINODE_ACCESS_TOKEN"]
        cluster_id ENV["KUBY_LINODE_CLUSTER_ID"]
      end
    end
  end
end
```

**Comment by camertron** [comment]
@mvz ah ok I see what the problem is. Just published v0.2.0 of docker-remote with the fix. Can you run `bundle update --source docker-remote` and try again?

**Comment by mvz** [comment]
Thanks, @camertron it works now. I'm having some issues later on in the deploy process but I'll open up a new ticket after I've done some debugging.
